### PR TITLE
feat(stats): add host_netinfo networking information stats family

### PR DESCRIPTION
**Pull request by ekoops**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

> /kind bug

> /kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

/kind feature

> /kind release

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area build

/area engine

> /area tests

> /area proposals

> /area CI

**What this PR does / why we need it**:

This PR introduces `host_netinfo` stats family to hold information regarding host networking. At the moment, it only provides ipv4 and ipv6 addresses lists for each interface available on the host.
The naming schema for the introduced stats is `falco.host_netinfo.interfaces.<ifname>.protocols.<ipv4|ipv6>.addresses="<address_list>"`, where `address_list` is a comma separated list of network addresses (e.g.: `"10.0.0.1,10.0.0.2"`). This schema avoids JSON parsing and is easily extensible (e.g.: we could easily introduce `ifindex` to specify the interface index, or `hw_addr` to specify the mac address).

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
feat(stats): add host_netinfo networking information stats family
```

### CODE REVIEW

File: userspace/falco/stats_writer.cpp [hunk 1]
```diff
@@ -309,6 +309,73 @@ stats_writer::collector::collector(const std::shared_ptr<stats_writer>& writer)
...
+	const std::shared_ptr<sinsp>& inspector)
+{
+	const auto ipv4_ifinfo = inspector->get_ifaddr_list().get_ipv4_list();
+	const auto ipv6_ifinfo = inspector->get_ifaddr_list().get_ipv6_list();
+
+	// For each interface name, collect the corresponding list of IPv4/IPv6 addresses
+	std::map<std::string, std::vector<std::string>> ifnames_to_ipv4_addresses;
+	std::map<std::string, std::vector<std::string>> ifnames_to_ipv6_addresses;
+
+	std::for_each(ipv4_ifinfo->cbegin(), ipv4_ifinfo->cend(), [&ifnames_to_ipv4_addresses](const auto& ifinfo) {
```
Comment (LucaGuerra) [inline userspace/falco/stats_writer.cpp:334]: I appreciate the "modern C++"ism but in the codebase we normally use regular for loops. If we have enough reasons to switch to algorithms we could do it throughout the codebase but I'd rather stick with what we use everywhere :)

Comment (ekoops) [reply userspace/falco/stats_writer.cpp:334]: Yes sure, no problem :)

File: userspace/falco/stats_writer.cpp [hunk 2]
```diff
@@ -309,6 +309,77 @@ stats_writer::collector::collector(const std::shared_ptr<stats_writer>& writer)
 {
 }
 
+std::string merge_strings(const std::vector<std::string>& v, const std::string& delim) {
```
Comment (LucaGuerra) [inline userspace/falco/stats_writer.cpp:312]: I believe we have utils somewhere to do this. We do have `sinsp_join()`, maybe we could use it here? Can you check if it works fine for your purpose?

**Comment by LucaGuerra** [comment]
I had just a couple minor comments, otherwise LGTM, thanks!

File: userspace/falco/stats_writer.cpp (hunk 2 above)
Comment (ekoops) [reply userspace/falco/stats_writer.cpp:312]: Yes I will check and update

**Review by FedeDP — APPROVED** [review]
I like it! Thanks!
cc @incertum 
/approve

**Comment by FedeDP** [comment]
@ekoops can you share an example output? Thanks!

**Comment by ekoops** [comment]
``````

**Review by LucaGuerra — APPROVED** [review]

**Pull request merged**